sidechain/top-pool-executor/src/call_operator.rs

```diff
@@ -0,0 +1,145 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{error::Result, TopPoolOperationHandler};
+use ita_stf::TrustedCallSigned;
+use itp_stf_executor::{
+	traits::{StfExecuteTimedCallsBatch, StfExecuteTimedGettersBatch},
+	BatchExecutionResult, ExecutedOperation,
+};
+use itp_time_utils::now_as_u64;
+use itp_types::{ShardIdentifier, H256};
+use its_primitives::traits::{Block as SidechainBlockT, SignedBlock as SignedBlockT};
+use its_state::{SidechainDB, SidechainState, SidechainSystemExt, StateHash};
+use its_top_pool_rpc_author::traits::{AuthorApi, OnBlockCreated, SendState};
+use log::*;
+use sgx_externalities::SgxExternalitiesTrait;
+use sp_runtime::{traits::Block as BlockT, MultiSignature};
+use std::{time::Duration, vec, vec::Vec};
+
+/// Interface to the trusted calls within the top pool
+pub trait TopPoolCallOperator {
+	type ParentchainBlockT: BlockT;
+
+	/// Loads trusted calls from the top pool for a given shard
+	/// and executes them until either all calls are executed or `max_exec_duration` is reached.
+	fn execute_trusted_calls(
+		&self,
+		latest_onchain_header: &<Self::ParentchainBlockT as BlockT>::Header,
+		shard: &ShardIdentifier,
+		max_exec_duration: Duration,
+	) -> Result<BatchExecutionResult>;
+
+	/// Retrieves trusted calls from the top pool.
+	fn get_trusted_calls(&self, shard: &ShardIdentifier) -> Result<Vec<TrustedCallSigned>>;
+
+	/// Removes the given trusted calls from the top pool.
+	/// Returns all hashes that were NOT successfully removed.
+	/// FIXME: Hash type should be taken from TrustedCall itself: #515
+	fn remove_calls_from_pool(
+		&self,
+		shard: &ShardIdentifier,
+		calls: Vec<ExecutedOperation>,
+	) -> Vec<ExecutedOperation>;
+}
+
+impl<PB, SB, RpcAuthor, StfExecutor> TopPoolCallOperator
```

Comment:
I have problems understanding the difference between executor and operator. Can you help me here? :)

Reply:
Executor executes something. However, with these changes it does not solely expose executions but more.